pallets/msa/src/lib.rs

```diff
@@ -207,10 +207,10 @@ pub mod pallet {
 		NotKeyOwner,
 		/// An operation was attempted with an unknown Key
 		NoKeyExists,
-		/// An operation was attempted with a revoked Key
-		KeyRevoked,
 		/// The number of key values has reached its maximum
 		KeyLimitExceeded,
+		/// The key is already registered to the MSA
+		KeyAlreadyRegistered,
 		/// A transaction's Origin (AccountId) may not revoke itself
```

Comment:
Is this a same as duplicatedKey?

Reply:
Not really
1. Duplicate  Key, if an msa who owns the key try to add it again
2. KeyAlreadyRegistered: if an account with different msa try to add a key which is already registered to another msa

Keeping a separation will ensure correct errors tell users whats wrong

Reply:
Maybe we just rename duplicatedKey to KeyAlreadyRegistered?

Reply:
KeyExists?

Reply:
I differ, we need a clear message whether this key is already added to my msa or this key is added to another msa

Think of range of users and providers and we need to have a clear demarcation of what is what 

KeyExists don't tell me If it exists for me or for someone else

Reply:
So we put some thought into the renaming and additional error. I like the idea of renaming DuplicatedKey to KeyAlreadyRegistered. However, we found that the added error might only benefit a very small niche of users.
This user is someone who manages multiple MSAs and want to know if he already registered this MSA or if I have assigned this Key other another MSA. Note that you can only register a key if you have proof that it belongs to you.
I think renaming the error KeyAlreadyRegistered will already tell the user that they have registered this key to another MSA and that the additional error does not add much value to this particular user since saying that the KeyAlreadyRegistered implies that they have already assigned this key to another MSA.



Duplicate -
The error will tell the user that some other MSA already registered this key. 

KeyAlreadyRegistered - 
The error will

Reply:
@enddynayn you are saying the opposite of what @saraswatpuneet was saying for the difference between the two errors. This is confusing enough that I think the naming is still not right.  One error is:
* This key is already registered to _this_ MSA.
* This key is already registered to _another_ MSA. 

The plain meaning of `KeyAlreadyRegistered` would apply to both cases; it doesn't make a distinction between them. `DuplicateKey` doesn't help draw the distinction.  So I think both names need improvement, perhaps:

`KeyAlreadyRegisteredToThisMSA`
`KeyAlreadyRegisteredToAnotherMSA`

Reply:
@johnmela  yes I like it we should have this separation between why this key is not getting added and having two set of errors gives correct information back to consumer

Reply:
@shannonwells  I liked this naming, this or something similar will be useful


I don't remember but we do have an issue in icebox or if not we should add to revisit our error names and update or make them more consumer friendly

Reply:
@enddynayn 's argument is that we don't need to make the distinction between the two cases of where a key has been already registered, This vs Another MSA. A holder of a private key should presumably know where said key has already been registered and must sign proof before an MSA can be created on their behalf. An RPC call to `get_msa_id` could return this information.

That being said, I don't see any detriment in us making this distinction with a clearer error message. I could see it being useful in the case where someone is managing multiple MSAs.  If we do keep the distinction then I also agree that we must make the distinction more clear in better error naming. @shannonwells 's suggestions of `KeyAlreadyRegisteredToThisMSA` vs. `KeyAlreadyRegisteredToAnotherMSA` seem clear enough to me. Since the distinction has already been pushed to this PR, we may as well rename errors.

Reply:
In the spirit of un complicating the error response and at @enddynayn 's request, I have merged the two error types into one and named it `KeyAlreadyRegistered`. I think this will likely be sufficient, we can revisit in the future if necessary.